### PR TITLE
edit to Application Basics page

### PR DESCRIPTION
**Pull request by sascala**

### CODE REVIEW

File: docs/docs/application-basics.md [hunk 1]
```diff
@@ -4,12 +4,13 @@ title: Application Basics
 
 # Application Basics
 
-Applications are an integral concept in Marathon. Each application typically represents a long running service, of which there would be many instances running on multiple hosts.
+Applications are an integral concept in Marathon. <!-- are they a concept? if so, what more can we say about this concept? --> Each application typically represents a long-running service, of which there would be many instances running on multiple hosts.
```
Comment (sascala) [inline docs/docs/application-basics.md]: Are applications truly a concept? If so, what more can we say about this concept?

File: docs/docs/application-basics.md [hunk 2]
```diff
@@ -21,19 +22,19 @@ You would use the following application definition (in JSON format) to describe
 }
 ```
 
-Note that `cmd` in the above example is the command that gets executed. Its value is wrapped by the underlying Mesos executor via `/bin/sh -c ${cmd}`.
+Note that `cmd` in the above example is the command that is executed. Its value is wrapped by the underlying Mesos executor via `/bin/sh -c ${cmd}`. <!-- this last sentence: why is it relevant to the user at this point? -->
```
Comment (sascala) [inline docs/docs/application-basics.md]: Why is this last sentence relevant to the user at this point?

File: docs/docs/application-basics.md [hunk 3]
```diff
@@ -21,19 +22,19 @@ You would use the following application definition (in JSON format) to describe
...
 
 <p class="text-center">
   <img src="{{ site.baseurl }}/img/marathon-basic-0.png" width="800" height="612" alt="Marathon deployment example: simple bash command">
 </p>
 
-What happens here is that Marathon hands over execution to Mesos. Mesos executes each task in its own sandbox environment.
-The sandbox is a special directory on each slave that acts as the execution environment (from a storage perspective) and also contains relevant log files as well as `stderr` and `stdout` for the command being executed. See also the role of the sandbox in [debugging distributed apps](https://docs.mesosphere.com/tutorials/debugging-a-mesosphere-cluster/).
+What happens here <!-- where is here? I want to rephrase this to "When you launch an app, Marathon... --> is that Marathon hands over execution to Mesos. Mesos executes each task in its own sandbox environment.
```
Comment (sascala) [inline docs/docs/application-basics.md]: Where is "here" in the first sentence? I want to rephrase this to "When you launch an app, Marathon...."

File: docs/docs/application-basics.md [hunk 4]
```diff
@@ -21,19 +22,19 @@ You would use the following application definition (in JSON format) to describe
...
+The sandbox is a special directory on each slave that acts as the execution environment (from a storage perspective <!-- what does this mean? -->) and contains relevant log files as well as `stderr` and `stdout` for the command being executed. See also the role of the sandbox in [debugging distributed apps](https://docs.mesosphere.com/tutorials/debugging-a-mesosphere-cluster/).
```
Comment (sascala) [inline docs/docs/application-basics.md]: What do we mean by "from a storage perspective?"

File: docs/docs/application-basics.md [hunk 5]
```diff
@@ -21,19 +22,19 @@ You would use the following application definition (in JSON format) to describe
...
 
 
 ## Using Resources in Applications
 
-To run any non-trivial application you typically depend on a collection of resources, i.e., files and/or archives of files. To deal with this, Marathon has the concept of `uris`. It leverages the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
+To run any non-trivial application, you typically depend on a collection of resources: files and/or archives of files. To deal with this <!-- I'd like to be more precise here: what is there to "deal with"? how can we describe the challenge here? -->, Marathon has the concept of `uris`. <!-- what is the connection between these two sentencences? --> Marathon leverages the Mesos fetcher to do the legwork in terms of downloading (and potentially) extracting resources.
```
Comment (sascala) [inline docs/docs/application-basics.md]: In the sentence that starts, "To deal with this..." I'd like to be more precise. What is there to "deal with"? how can we describe the challenge here?

File: docs/docs/application-basics.md [hunk 6]
```diff
@@ -50,11 +51,11 @@ But before we dive into this topic, let's have a look at an example:
 }
 ```
 
-What the example above does: before executing the `cmd`, download the resource `https://example.com/app/cool-script.sh` (via Mesos) and make it available in the application task's sandbox. You can check that these have been downloaded by visiting the Mesos UI and clicking into a Mesos worker node's sandbox where you should now find `cool-script.sh`.
+The example above downloads the resource `https://example.com/app/cool-script.sh` (via Mesos) and makes it available in the application task's sandbox before executing the `cmd`. You can check that these have been downloaded by selecting the sandbox of a Mesos worker node in the Mesos UI. You should find `cool-script.sh`.
 
-Note that as of Mesos v0.22 and above the fetcher code now does not make downloaded files executable by default, so `cmd` should be: `chmod u+x cool-script.sh && ./cool-script.sh`.
+Note that as of Mesos v0.22 and above the fetcher code does not make downloaded files executable by default, so `cmd` should be: `chmod u+x cool-script.sh && ./cool-script.sh`. <!-- should we just have this be cmd, then comment on it? wouldn't most people be using the latest Mesos? -->
```
Comment (sascala) [inline docs/docs/application-basics.md]: Should we just have this be the value of cmd, then comment on it? Would most people be using the latest Mesos?

File: docs/docs/application-basics.md [hunk 7]
```diff
@@ -107,9 +108,9 @@ A typical pattern in the development and deployment cycle is to have your automa
...
+It is straightforward to run applications that use Docker images in Marathon. See [Running Docker Containers on Marathon]({{ site.baseurl }}/docs/native-docker.html) for further details and advanced options.
 
-In the following example application definition, we will focus on a simple Docker app: a Python-based web server using the image [python:3](https://registry.hub.docker.com/_/python/). Inside the container, the web server runs on port `8080` (the value of `containerPort`). Outside of the container, Marathon assigns a random port (`hostPort` is set to `0`):
+In the following example application definition, we will focus on a simple Docker app: a Python-based web server using the image [python:3](https://registry.hub.docker.com/_/python/). Inside the container, the web server runs on port `8080` (the value of `containerPort`). Outside of the container, Marathon assigns a random port (`hostPort` is set to `0`): <!-- does this mean that if you set hostPort to 0 you'll get assigned a random port? -->
```
Comment (sascala) [inline docs/docs/application-basics.md]: Does the last sentence mean that if you set hostPort to 0 you'll get assigned a random port?

File: docs/docs/application-basics.md (hunk 1 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: Perhaps there is a better term? An application definition describes everything to start and maintain tasks.

File: docs/docs/application-basics.md (hunk 2 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: Because you can rely on the shell behavior like e.g. environment variable expansion.

File: docs/docs/application-basics.md (hunk 3 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: I think the sentence can be improved. It says: when this app is defined in Marathon, Marathon will launch launch tasks via Mesos.

File: docs/docs/application-basics.md (hunk 4 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: Please rewrite. The sandbox is a directory that is created by Mesos for each task. The `stdout` and `stderr` stream is written to this location.

File: docs/docs/application-basics.md (hunk 5 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: Every task needs some resources to run: Memory, CPU, Disk etc.

File: docs/docs/application-basics.md (hunk 6 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: I think so.

File: docs/docs/application-basics.md (hunk 7 above)
Comment (aquamatthias) [inline docs/docs/application-basics.md]: Yes.

**Comment by aquamatthias** [comment]
Close for now - please reopen when ready.
